packages/app/background-jobs-common/src/background-job-processor/types.ts

```diff
@@ -1,15 +1,18 @@
 import type { FinishedStatus, Job, JobsOptions, Queue } from 'bullmq'
 
-import type { RequestContext } from './processors/AbstractBackgroundJobProcessor'
+import type { CommonLogger } from '@lokalise/node-core'
+
+export interface RequestContext {
+  logger: CommonLogger
+  reqId: string
+}
 
 export type JobFinalState = FinishedStatus
 export type BaseJobPayload = { metadata: { correlationId: string } }
 
 // "scripts" field is incompatible between free and pro versions, and it's not particularly important
 // biome-ignore lint/suspicious/noExplicitAny: it's okay
-export type SafeJob<T = any, R = any, N extends string = string> = Omit<Job<T, R, N>, 'scripts'> & {
-  requestContext?: RequestContext
-}
+export type SafeJob<T = any, R = any, N extends string = string> = Omit<Job<T, R, N>, 'scripts'>
```

Comment:
Removing `requestContext` from `SafeJob` so it is not visible for child classes, handling it in a different way but keeping `requestContext` on the job object to avoid recreation if not needed